projects/plugins/jetpack/modules/carousel/jetpack-carousel.php

```diff
@@ -357,10 +335,133 @@ function enqueue_assets() {
 			 */
 			do_action( 'jp_carousel_enqueue_assets', $this->first_run, $localize_strings );
 
+			// Add the carousel skeleton to the page.
+			$this->localize_strings = $localize_strings;
+			add_action( 'wp_footer', array( $this, 'add_carousel_skeleton' ) );
+
 			$this->first_run = false;
 		}
 	}
 
+	/**
+	 * Generate the HTML skeleton that will be picked up by the Carousel JS and used for showing the carousel.
+	 */
+	public function add_carousel_skeleton() {
+		$localize_strings = $this->localize_strings;
+		$is_light         = ( 'white' === $localize_strings['background_color'] );
+		// Determine whether to fall back to standard local comments.
+		$use_local_comments = ! isset( $localize_strings['jetpack_comments_iframe_src'] ) || empty( $localize_strings['jetpack_comments_iframe_src'] );
+		$current_user       = wp_get_current_user();
+		$require_name_email = (int) get_option( 'require_name_email' );
+		/* translators: %s is replaced with a field name in the form, e.g. "Email" */
+		$required = ( $require_name_email ) ? __( '%s (Required)', 'jetpack' ) : '%s';
+		?>
+
+		<div
+			class="jp-carousel-wrap jp-carousel-transitions<?php echo( $is_light ? ' jp-carousel-light' : '' ); ?>"
+			itemscope
+			itemtype="https://schema.org/ImageGallery"
+			style="display: none;">
+			<div class="jp-carousel-overlay"></div>
+			<div class="jp-carousel"></div>
+			<div class="jp-carousel-fadeaway"></div>
+			<div class="jp-carousel-info">
+				<div class="jp-carousel-photo-info">
+					<h2 class="jp-carousel-caption" itemprop="caption description"></h2>
+				</div>
+				<div class="jp-carousel-info-columns">
+					<div class="jp-carousel-left-column-wrapper">
+						<div class="jp-carousel-titleanddesc"></div>
+						<!-- Intentional duplicate -->
+						<div class="jp-carousel-photo-info">
+							<h2 class="jp-carousel-caption" itemprop="caption description"></h2>
+						</div>
+						<?php if ( $localize_strings['display_comments'] ) : ?>
+							<div id="jp-carousel-comment-form-container">
+								<?php if ( $use_local_comments ) : ?>
+									<?php if ( ! $localize_strings['is_logged_in'] && $localize_strings['comment_registration'] ) : ?>
+										<div id="jp-carousel-comment-form-commenting-as">
+											<p id="jp-carousel-commenting-as">
+												<?php // phpcs:ignore Standard.Category.SniffName.ErrorCode,WordPress.Security.EscapeOutput.UnsafePrintingFunction ?>
+												<?php _e( 'You must be <a href="#" class="jp-carousel-comment-login">logged in</a> to post a comment.', 'jetpack' ); ?>
```

Comment:
Could we do something like this instead?

```suggestion
												<?php
													echo wp_kses(
														__( 'You must be <a href="#" class="jp-carousel-comment-login">logged in</a> to post a comment.', 'jetpack' ),
														array(
															'a' => array(
																'href'  => array(),
																'class' => array(),
															),
														)
													);
												?>
```

Reply:
Thanks, TIL!